### ISSUE TITLE
Incorrent error message (ts1166) when symbol is in fact a literal unique symbol given by `Symbol.for('wellknownname')`

### ISSUE THREAD

**Issue by craigphicks** [microsoft/TypeScript#42601]
# Bug Report

### 🔎 Search Terms

"A computed property name in a class property declaration must refer to an expression whose type is a literal type or a 'unique symbol' type. "

"ts(1166)"

### 🕗 Version & Regression Information

- I was unable to test this on prior versions because unlikely to be any different

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/MYewdgzgLgBBC8BlAngWwEYgDYAoDkAZiCHgJQBQoksEBSamWAdEQE75EkXBYCGEEGAGEYAb3Iw4UXlACWwGKwCmvACbgsyGAG1CxPAF0AXACUloVqoA80VrLABzADS8wyAHwSpM+YpXqwTR0IYxgzC2tbe2dXDy9oHwVlNQ0tbVpQ8JBLGyg7Rxc3T0kEuST-VJ0UDGwWbI59UkzzbMi86MK4gF9yAHpemABZLVBVJRgHECVBAAslZSA)

### 💻 Code

```
const s=Symbol('foo')
const sf=Symbol.for('foo')
class C {
  static readonly ['foo']:Record<string,any>
  static readonly [s]: Record<string,any>
  static readonly [sf]: Record<string,any>
  static readonly [Symbol.for('foo')]: Record<string,any>     // <---- oxymoronic error here   (ts1166)
}
```
Error:
```
A computed property name in a class property declaration must refer to an expression whose type is a literal type or a 'unique symbol' type.  (ts1166)
```

### 🙁 Actual behavior

Fatal compile error

### 🙂 Expected behavior

Pass, because the symbol is in fact a unique (constant) symbol.

I guess that [when well known symbols were implemented in 2014](https://github.com/microsoft/TypeScript/issues/980) `Symbol()` was the only way to create a symbol, and `Symbol.for()` did not exist. Therefore the only practical way to use a well known symbol as an index was to pass a constant reference.  However, `Symbol.for()` is now available, the most direct and least verbose way to use a "custom" well known symbol as an index is to write it directly as the index, as shown in the example - no imports required.  
  To put it another way, the argument to `Symbol.for(argument)` is an import-like statement for the symbol, the argument being well-known from documentation for the application/tool employing it.

Just to clarify, I am not suggesting changes for existing well symbols such as the iterator symbol.  Only new application specific custom uses.

**Comment by ajafff** [comment]
The error message is in fact very misleading. This is not specific to symbols, but also applies to literal types.
The compiler expects a variable or "dotted name" (property access on variable or another property access, ...) with a literal or unique symbol type. In your case there's a call expression, which is not allowed by the current check (for reasons I guess?), so this has nothing to do with the return type of the call.

Consider this code:

```ts
declare var o: {prop: 'foo'};

class C {
    [o.prop] = 1; // works
    [(o).prop] = 1; // adding parens gives an error because it's syntactically no longer a dotted name
}
```

**Comment by craigphicks** [comment]
@ajafff 

I interpret what you are saying to imply that:
    This error is generated at the AST parsing stage, and not the symbol binding stage.

I understand that performance reasons it is not currently possible to either:
1. Postpone all error processing until the binding stage
2.  Handle every possible syntactic indirection, e.g., your parenthesis.  (That's done during the binding stage anyway.)

I think it fair to say that adding parenthesis is a "meaningless" indirection, and there is no reason to allow it.

In contrast, there is *less* indirection in 
```
class C {
   static readonly [Symbol.for('specialSymbolKeyblahblah')]
}
```
compared to 
```
const specialSymbol=Symbol.for('specialSymbolKeyblahblah')
class C {
   static readonly [specialSymbol]
}
```

On the other hand, typing an unchecked string into `Symbol.for('uncheckedString')`, is probably not a good idea from the point of view of spelling mistakes.  To avoid that would require at least something like the following: 

```
import {specialSymbolName}  from 'xxxxx'  // export const specialSymbolName='specialSymbolNameblahblah'
class C {
   static readonly [Symbol.for(specialSymbolName)]
}
```
which doesn't offer any more (and possibly less) convenience than
```
import {specialSymbol}  from 'xxxxx'  // export const specialSymbol=Symbol('informal mnemonic')
class C {
   static readonly [specialSymbol]
}
```
For that reason I will close this issue.  

I would, however, recommend changing the error message from
```
"A computed property name in a class property declaration must refer to an expression whose type is a literal type or a 'unique symbol' type. "
```
to
```
"A computed property name in a class property declaration must refer to an explicit expression for a literal string or a literal number, or a constant reference to a symbol. "
```


> The error message is in fact very misleading. This is not specific to symbols, but also applies to literal types.
> The compiler expects a variable or "dotted name" (property access on variable or another property access, ...) with a literal or unique symbol type. In your case there's a call expression, which is not allowed by the current check (for reasons I guess?), so this has nothing to do with the return type of the call.
> 
> Consider this code:
> 
> ```ts
> declare var o: {prop: 'foo'};
> 
> class C {
>     [o.prop] = 1; // works
>     [(o).prop] = 1; // adding parens gives an error because it's syntactically no longer a dotted name
> }
> ```
